### PR TITLE
VPN Endpoint V2

### PR DESCRIPTION
**Pull request by mbfrahry**
This PR adds VPN Endpoint V2 to the sdk. It's going to take a little more though to test this as we need an existing VPN Endpoint to get it working

### CODE REVIEW

File: compute/vpn_endpoint_v2.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+			ResourceDescription: VPNEndpointV2Description,
+			ContainerPath:       VPNEndpointV2ContainerPath,
+			ResourceRootPath:    VPNEndpointV2ResourcePath,
+		},
+	}
+}
+
+// VPNEndpointV2Info contains the exported fields necessary to hold all the information about an
+// VPN Endpoint V2
+type VPNEndpointV2Info struct {
```
Comment (scross01) [inline compute/vpn_endpoint_v2.go]: The API also returns a `lifecycleState` attribute, which doesn't appear to be in the docs, but we'll need to monitoring the provisioning, update and delete state of the VPN endpoint with a "wait for ready" function, initial VPN endpoint provisioning can take >1 hour.  The values for the `lifecycleState` are: 

- `provisioning` - VPN endpoint is provisioning (initial state on create)
- `ready` - VPN endpoint provisioned
- `updating` - VPN endpoint is being updated (after update request)
- `deleting`  - VPN endpoint is being deleted (after delete request)
- `error` - VPN endpoint provisioning failed (error state)

In addition, the update and delete will operations can be run when the VPN endpoint is in a `ready` or `error` lifecycle state.

**Comment by scross01** [comment]
For the automated testing of the VPN endpoint resource an existing remote-end endpoint is not necessary - the testing just needs to check that the lifecycle state gets to `ready`, which is independent of the tunnel state.  

Testing the full tunnel setup can be done by setting two endpoints in two different regions (e.g uscom-central-1 and uscom-east-1) however the setup is tricky to automate as the each end needs to know the details of the other end at creation time, to get around the chicken and egg situation the steps involve bringing up the first endpoint with dummy remote connection settings, bring up the second endpoint to connect to the first, and then update the first endpoint with the correct details for the second endpoint.

File: compute/vpn_endpoint_v2.go (hunk 1 above)
Comment (mbfrahry) [reply compute/vpn_endpoint_v2.go]: This has been added

File: compute/vpn_endpoint_v2_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+	createInput := &CreateVPNEndpointV2Input{
+		Name:               _VPNEndpointV2TestName,
+		CustomerVPNGateway: _VPNEndpointV2TestCustomerVPNGateway,
+		IPNetwork:          ipNetwork.Name,
+		PSK:                _VPNEndpointV2TestPSK,
+		ReachableRoutes:    []string{"192.168.155.2/24"},
+	}
+
+	// Create a vNIC Set
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply compute/vpn_endpoint_v2_test.go]: We can specify `default`instead of a creating a vnic set

**Comment by mbfrahry** [comment]
I can create an endpoint with a dummy remote connection. Unfortunately, it doesn't look like the `enabled` flag works for updating an endpoint

File: compute/vpn_endpoint_v2_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (scross01) [reply compute/vpn_endpoint_v2_test.go]: I think the vnic set needs the fully qualified name as the UI shows and error the `default` vNICset does not exist, however as the test is not actually establishing a connected tunnel I don't think it matters as the test confirms the value is set as provided.

**Comment by scross01** [comment]
LGTM

**Pull request merged**